api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1036,14 +1037,14 @@ private HttpResponse executeDelete(String bucketName, String objectName, Map<Str
    * @param data           HTTP request body data.
    */
   private HttpResponse executePost(String bucketName, String objectName, Map<String,String> headerMap,
-                                   Map<String,String> queryParamMap, Object data)
+                                   Map<String,String> queryParamMap, String contentType, Object data)
```

Comment:
1. As `headerMap` already carries `Content-Type`, we would need to use it.
2. You could remove `contentType` argument in `execute()` and get `contentType` from `headerMap` and pass it to `createRequest()` or `createRequest()` could get `contentType` from `headerMap`